pkg/bucketeer/model/status.go

```diff
@@ -117,3 +119,31 @@ func NewServiceUnavailableErrorMetricsEvent(tag string, api APIID) *ServiceUnava
 		Type:   ServiceUnavailableErrorMetricsEventType,
 	}
 }
+
+type PayloadTooLargeErrorMetricsEvent struct {
+	APIID  APIID                `json:"apiId,omitempty"`
+	Labels map[string]string    `json:"labels,omitempty"`
+	Type   errorStatusEventType `json:"@type,omitempty"`
+}
+
+func NewPayloadTooLargeErrorMetricsEvent(tag string, api APIID) *PayloadTooLargeErrorMetricsEvent {
+	return &PayloadTooLargeErrorMetricsEvent{
+		APIID:  api,
+		Labels: map[string]string{"tag": tag},
+		Type:   PayloadTooLargeErrorMetricsEventType,
+	}
+}
+
+type RedirectionRequestErrorMetricsEvent struct {
+	APIID  APIID                `json:"apiId,omitempty"`
+	Labels map[string]string    `json:"labels,omitempty"`
+	Type   errorStatusEventType `json:"@type,omitempty"`
+}
+
+func NewRedirectionRequestErrorMetricsEvent(tag string, api APIID) *RedirectionRequestErrorMetricsEvent {
+	return &RedirectionRequestErrorMetricsEvent{
+		APIID:  api,
+		Labels: map[string]string{"tag": tag},
```

Comment:
We should also implement the `response_code` to report the code.

Reference:
https://github.com/bucketeer-io/ios-client-sdk/pull/68/files#diff-c3220166f7e54cc576f13734d3d0d8f3d60a8e543e2fd9a56e4b7ea100a0f643R360-R363

https://github.com/bucketeer-io/ios-client-sdk/pull/68/files#diff-21e1c84fd4bacc9267f4a4cd72b9f6a4a37dd6ab724b13ee0d18973da4b186b9R169-R171